scipy/stats/_fit.py

```diff
@@ -688,3 +689,541 @@ def nllf(free_params, data=data):  # bind data NOW
+    fitted parameter values. To overcome these shortcomings, this function
+    allows the user to perform the Monte Carlo trials adapted to their
+    particular data.
+
+    First, any unknown parameters of the distribution family specified by
+    `dist` are fit to the provided `data`. These values of the parameters
+    specify a particular member of the distribution family referred to as the
+    "null-hypothesized distribution", that is, the distribution from which the
+    data were sampled under the null hypothesis. The `statistic`, which
+    compares data to a distribution, is computed between `data` and the
+    null-hypothesized distribution.
+
+    Next, many (specifically `n_mc_samples`) new samples, each containing the
+    same number of observations as `data`, are drawn from the
+    null-hypothesized distribution. Any unknown parameters of the distribution
+    family `dist` are fit to *each resample*, and the `statistic` is computed
+    between each sample and its corresponding fitted distribution. These
+    values of the statistic form the Monte Carlo null distribution (not to be
+    confused with the "null-hypothesized distribution" above).
+
+    The p-value of the test is the proportion of statistic values in the Monte
+    Carlo null distribution that are at least as extreme as the statistic value
+    of the provided `data`. More precisely, the p-value is given by
+
+    .. math::
+
+        p = \frac{b + 1}
+                 {m + 1}
+
+    where :math:`b` is the number of statistic values in the Monte Carlo null
+    distribution that are greater than or equal to the the statistic value
+    calculated for `data`, and :math:`m=` is the number of elements in the
+    Monte Carlo null distribution (`n_mc_samples`). The addition of :math:`1`
+    to the numerator and denominator can be thought of as including the
+    value of the statistic corresponding with `data` in the null distribution,
+    but a more formal explanation is given in [5]_.
+
+    The test can be very slow for some distribution families because unknown
+    parameters of the distribution family must be fit to each of the Monte
+    Carlo samples, and for most distributions in SciPy, distribution fitting
+    performed via numerical optimization. For this reason, it may be tempting
+    to treat parameters of the distribution pre-fit to `data` (by the user)
+    as though they were `known_params`, as specification of all parameters of
+    the distribution precludes the need to fit the distribution to each Monte
+    Carlo sample. (This is essentially how the original Kilmogorov-Smirnov
+    test is performed.) Although such a test can provide evidence against the
+    null hypothesis, the power of the test is low (that is, it is less likely
+    to reject the null hypothesis even when the null hypoothesis is false)
+    because the Monte Carlo samples are less likely to agree with the
```

Comment:
@chrisb83 does this help?
```suggestion
    null hypothesis, the test is conservative in the sense that small p-values 
    will tend to (greatly) *overestimate* the probability of making a type I
    error (that is, rejecting the null hypothesis although it is true), and the 
    power of the test is low (that is, it is less likely to reject the null 
    hypothesis even when the null hypothesis is false).
    This is because the Monte Carlo samples are less likely to agree with the
```